### DISCUSSION TITLE
Module 'machine' not found in Pico-W-Go/VS Code

### DISCUSSION THREAD

**Discussion by nick6502** [micropython/micropython#11364] [RP2040 / Pico]
I'm fairly new to micropython and the pi pico but having some success.  

Current problem:  I'm trying to use VS Code with Pico-W-Go, which is installed and working, more or less. But when I try this sample program which came with Pico-W-Go:

```python
from machine import Pin
from utime import sleep

pin = Pin(25, Pin.OUT)

print("LED starts flashing...")
while True:
    pin.toggle()
    sleep(1) # sleep 1sec

```

I get the error "ModuleNotFoundError: No module named 'machine'"

It's interesting though, that programs run under Thonny which invoke 'machine' work OK.  How can I import machine into VS Code/Pico-W-Go, or tell it where to find machine?  I'm using Windows 11, BTW.

Nick

**Comment by gregwa1953** [comment]
According to https://forums.raspberrypi.com/viewtopic.php?t=35100 , 

> That's because all the standard library modules are packed within the micropython binary (.uf2) file you flashed to the Pico to install micropython and are not part of the filesystem micropython exposes to the USB host over the REPL.

So you SHOULD be able (in theory) to do this...
`led_pin = machine.Pin(25, machine.Pin.OUT)
`
and the "machine" portion has meaning, because it is magically imported by default without the need of implicitly being imported.

Gregwa

**Reply by Josverl** [reply]
Installing the micropython-stubs in a venv should help your IDE understand the ons and outs of micropython.

https://micropython-stubs.readthedocs.io/en/main/

**Reply by edgexie** [reply]
I already installed the stubs but the IDE notice me “can't import `machine` from analysis ”.

![image](https://github.com/micropython/micropython/assets/18474752/b492062b-91b6-4b61-ab51-6232e9060402)

**Comment by nick6502** [comment]
Thanks Gregwa,

I tried that and got the same error.  Is it possible that the micropython I
installed while installing Thonny is different from a micropython I might
have installed if I'd started with VS Code?

Thanks,

Nick

**Comment by bixb922** [comment]
That is not a Micropython error but a Python error. Your IDE is connected to Python instead of Micropython The error would look this way in Micropython:
```
>>> import other
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ImportError: no module named 'other'
```
You have to import machine to use Pin, the code you posted is correct.

**Comment by gregwa1953** [comment]
I'm just wondering...What version of Thonny are you using?  Mine is 4.0.2 .

I copied the RPi Forum code to my machine and ran it on my Pico-w and it worked fine.

```
import utime

# HOW is the next line working without 'import machine'?
led_pin = machine.Pin("LED", machine.Pin.OUT)

led_pin.value(1)
utime.sleep(3)
led_pin.value(0)
```

Then I ran yours with the Pin changed from 25 to "LED" for the internal and ran it.  Again, it worked just fine.


```
from machine import Pin
from utime import sleep

pin = Pin("LED", Pin.OUT)

print("LED starts flashing...")
while True:
    pin.toggle()
    sleep(1) # sleep 1s
```

Greg

**Reply by scruss** [reply]
> HOW is the next line working without 'import machine'?

Because [_boot.py](https://github.com/micropython/micropython/blob/master/ports/rp2/modules/_boot.py) for the rp2 contains `import machine`

**Reply by gregwa1953** [reply]
Thank you @scruss .
That helps!

**Comment by scruss** [comment]
> ```python
> pin = Pin(25, Pin.OUT)
> ```

isn't going to work anyway. The recommended syntax is

> ```python
> pin = Pin('LED', Pin.OUT)
> ```

**Comment by lastant** [comment]
I also had this error, because I was pressing F5 (Run->Start Debugging), and apparently one is supposed to click on Run in the status bar at the bottom of the screen or Ctrl+Shift+P -> "Pico-W-Go: Run current file on Pico"

**Comment by scruss** [comment]
> I get the error "ModuleNotFoundError: No module named 'machine'"

What version of MicroPython did you install?

If you accidentally installed CircuitPython, it's incompatible, and doesn't have a `machine` module
